src/Types.ts

```diff
@@ -24,11 +24,15 @@ export enum TokenStandard {
 
 export type BlockState = 'MINED' | 'NOT_MINED'
 
-export type NonceState = 'USED' | 'NOT_USED'
+export const USED_NONCE_STATE = 'USED';
+export const NOT_USED_NONCE_STATE = 'NOT_USED';
+export type NonceState = typeof USED_NONCE_STATE | typeof NOT_USED_NONCE_STATE;
 
```

Comment:
@phanpak this seems a lot more complicated than just using strings. any reason why we need to do this?

Reply:
@phanpak would this be equivalent?
```
export enum NonceState {
    USED,
    NOT_USED
}
```